datasets/um005/um005.py

```diff
@@ -0,0 +1,137 @@
+"""
+_HOMEPAGE_URL = "http://ufal.ms.mff.cuni.cz/umc/005-en-ur/"
+_URL = "http://ufal.ms.mff.cuni.cz/umc/005-en-ur/download.php?f=umc005-corpus.zip"
+_CITATION = """\
+@unpublished{JaZeWordOrderIssues2011,
+author      = {Bushra Jawaid and Daniel Zeman},
+title       = {Word-Order Issues in {English}-to-{Urdu} Statistical Machine Translation},
+year        = {2011},
+journal     = {The Prague Bulletin of Mathematical Linguistics},
+number      = {95},
+institution = {Univerzita Karlova},
+address     = {Praha, Czechia},
+issn        = {0032-6585},
+}
+"""
+
+_ALL = "all"
+_VERSION = "1.0.0"
+_SOURCES = ["bible", "quran"]
+_SOURCES_FILEPATHS = {
+    s: {
+        "train": {"urdu": "train.ur", "english": "train.en"},
+        "dev": {"urdu": "dev.ur", "english": "dev.en"},
+        "test": {"urdu": "test.ur", "english": "test.en"},
+    }
+    for s in _SOURCES
+}
+
+
+class UM005Config(datasets.BuilderConfig):
+    def __init__(self, *args, sources=None, **kwargs):
+        super().__init__(*args, version=datasets.Version(_VERSION, ""), **kwargs)
+        self.sources = sources
+
+    @property
+    def filepaths(self):
+        return {source: _SOURCES_FILEPATHS[source] for source in self.sources}
+
+
+class UM005(datasets.GeneratorBasedBuilder):
+    BUILDER_CONFIGS = [
+        UM005Config(name=source, sources=[source], description=f"Source. {source}.") for source in _SOURCES
+    ] + [UM005Config(name=_ALL, sources=_SOURCES, description=f"All sources included.")]
+
+    DEFAULT_CONFIG_NAME = _ALL
+
+    def _info(self):
+        return datasets.DatasetInfo(
+            description=_DESCRIPTION,
+            features=datasets.Features(
+                {
+                    "id": datasets.Value("string"),
+                    "urdu": datasets.Value("string"),
+                    "english": datasets.Value("string"),
+                },
+            ),
+            supervised_keys=None,
+            homepage=_HOMEPAGE_URL,
+            citation=_CITATION,
+        )
+
+    def _split_generators(self, dl_manager):
+        path = dl_manager.download_and_extract(_URL)
+        return [
+            datasets.SplitGenerator(
+                name=datasets.Split.TRAIN,
+                gen_kwargs={"datapath": path, "datatype": "train"},
+            ),
+            datasets.SplitGenerator(
+                name=datasets.Split.VALIDATION,
+                gen_kwargs={"datapath": path, "datatype": "dev"},
+            ),
+            datasets.SplitGenerator(
+                name=datasets.Split.TEST,
+                gen_kwargs={"datapath": path, "datatype": "test"},
+            ),
+        ]
+
+    def _generate_examples(self, datapath, datatype):
+        sentence_counter = 0
+        for source, filepath in self.config.filepaths.items():
+            filepath = filepath[datatype]
+            urdu_file = filepath["urdu"]
+            english_file = filepath["english"]
+            urdu_path = os.path.join(datapath, source, urdu_file)
```

Comment:
would be nice if the "path" creation could be done in `_split_generators`. I'd just pass the two filepaths or two lists of filepaths if '"all"' is needed.